applications/dashboard/modules/class.dropdownmodule.php

```diff
@@ -232,4 +238,45 @@ public function setTrigger($text = '', $type = 'button', $cssClass = 'btn-defaul
         $this->trigger['url'] = $url;
         return $this;
     }
+    /**
+     * Returns array of translations for dropdown menu
+     *
+     * @return array
+     */
+    protected static function tInit() {
+        return [
+            'Dismiss' => Gdn::translate('Dismiss'),
+            'Edit' => Gdn::translate('Edit'),
+            'Announce' => Gdn::translate('Announce'),
+            'Unsink' => Gdn::translate('Unsink'),
+            'Sink' =>  Gdn::translate('Sink'),
+            'Reopen' => Gdn::translate('Reopen'),
+            'Close' => Gdn::translate('Close'),
+            'Refetch Page' => Gdn::translate('Refetch Page'),
+            'Move' => Gdn::translate('Move'),
+            'Tag' => Gdn::translate('Tag'),
+            'Delete Discussion' => Gdn::translate('Delete Discussion'),
+            'Options' => Gdn::translate('Options')
+        ];
+    }
+
+    /**
+     * Method overwrites parent method to cache response statically
+     *
+     * @param string $view Template name
+     * @param string $applicationFolder App folder to check for template
+     *
+     * @return array|mixed
+     *
+     * @throws Exception Exception could be thrown by parent::fetchViewLocation() when template is not found.
+     *
+     */
+    public function fetchViewLocation($view = '', $applicationFolder = '') {
```

Comment:
Yes it can. But if you see we are always kind have opposite opinions. I want everything to be applied at very end when we need. imho that is the most efficient way to isolate problems and keep code clean and efficient. - whenever we need to get rid of dependency -we can just remove `use Trait;` form the child class. I don't want to implement many traits and interfaces on parent core class. Which will be never used by most children classes.

Reply:
> I want everything to be applied at very end when we need.

My take on that is if we can effectively implement a simple optimization in a way that is easily shared, we should do that. Only applying optimizations to single classes as we need to is going to be tedious and has a much higher risk of code dupe. Unless I'm reviewing every module in the application, how would I know this method exists for use in a new module? There's nothing in the parent class. There's not traits out there. I'd end up just re-building this caching mechanism, when a shared mechanism would've been a better option.

Reply:
Yes, I see your point. But unfortunately disagree with your arguments. :-) Looking on the same code we have opposite impressions and different approaches of how we should go about the problem. 

Right now I do optimize some "home page" view with some number of discussions in the feed. I do profiling and I see loops doing the same all the time. To optimize the code - I do some kind local modifications (as close to the output view layer as I can) which allows me to feel safe and secure. To be sure that I don't break any other view/model/module/plugin which I have no idea even exist. 

I would feel very uncomfortable to apply something globally. When every module/view/controller/plugin (I have no idea about complexity on production) will get new method/property I will feel/get all responsibility if something is wrong and/or unexpected happens. This code was never tested by any QA. And I don't see how I can test it before getting to production if there is any side effects on memory or CPU usage.

I prefer to apply same tricks few times on some particular classes. To get positive results on production. To experience some issues probably with some particular clients on these few classes. After that when I see that this trick is working pretty good in particular cases I would extend it and try to move one layer up (probably to some base model or base controller or base plugin). But not take this risk right now. IMHO if something is working fine on my Dev it does not mean it will work even on any developer dev. If that works fine on staff.vanillaforum - that does not mean it will work everywhere. I am not that confident as probably you are.

I would be glad to put all my ideas into core one day. But not now. Not something that was never tested on production.

Reply:
> I would feel very uncomfortable to apply something globally.

I'm not suggesting this be used by all modules by rewriting the parent's `fetchViewLocation` method. I'm suggesting the caching logic be moved into a separate method on the parent class or into a trait--something reusable. Duplicate code is code that is difficult to maintain. You could make a call to the caching method here, when this method is overridden. Other modules would then be able to selectively implement this, without the need for "globally"  applying anything.

Reply:
As I said, I see your point. But disagree in this particular case.
My intention is to test some ideas. I always try to avoid assumptions about "shared" part of the code. You call it dupes/duplicates. I don't.

To me duplicates are when I copy-paste it several times a day/week/month and remember that I did it recently - that means - I need to do some research and probably create something to share between some classes. But that is not something obvious to me and not something I would force. If I see `echo` five times in consecutive lines - probably there is something to set to separate method but maybe not.

In our case - that is 1st time I apply my cache to the view. I have no idea if that is good or bad idea. I have no idea how this change will affect all our themes and plugins on production. I want to keep change as local as I can. I don't want to resolve all issues at once. All future improvements for other cases even for you they are very predictable - to me they are not or they do not exist yet.

I want to explain more to you just to explain my vision as much as I can. So you can understand my intentions and my way to develop, my habits and etc.